### PR TITLE
abstract build configuration to clean up api

### PR DESCRIPTION
**Pull request by DarylThayil**
Moving build config from a public to a private function to clean up the user experience and make configuring an app easier.

Still to be done before merging
[x] write more tests

### CODE REVIEW

File: lib/msal-core/samples/VanillaJSTestApp/index_LoginPopup.html [hunk 1]
```diff
@@ -46,16 +46,19 @@ <h1>Sending an email with msal.js and Microsoft Graph</h1>
...
-    // Setting a default value in JSON format
-    var config = Msal.buildConfiguration(
-      { clientId: applicationConfig.clientID, validateAuthority: true, },
-      { cacheLocation: "localStorage", storeAuthStateInCookie: true },
-      { logger: logger },
-      {}
-    );
-
-    var userAgentApplication = new Msal.UserAgentApplication(config);
+    var userAgentApplication = new Msal.UserAgentApplication({
```
Comment (DarylThayil) [inline lib/msal-core/samples/VanillaJSTestApp/index_LoginPopup.html]: define config outside msal

File: lib/msal-core/src/Configuration.ts [hunk 2]
```diff
@@ -105,58 +132,13 @@ export type Configuration = {
...
-      logger,
-      loadFrameTimeout,
-      tokenRenewalOffsetSeconds
-    },
-    framework: {
-      isAngular,
-      unprotectedResources,
-      protectedResourceMap
-    }
+export function buildConfiguration({ auth, cache={}, system={}, framework={}}: Configuration): Configuration {
```
Comment (sameerag) [inline lib/msal-core/src/Configuration.ts]: The above code can cause linter error for us, cache = {} could be the right way. Minor but edit if you can.

File: lib/msal-core/samples/VanillaJSTestApp/index_LoginPopup.html (hunk 1 above)
Comment (sameerag) [reply lib/msal-core/samples/VanillaJSTestApp/index_LoginPopup.html]: +1 Looks much better.

File: lib/msal-core/package.json [hunk 3]
```diff
@@ -33,9 +33,7 @@
     "lint": "tslint -c tslint.json src/**/*.ts",
     "build:webpack": "webpack"
   },
-  "dependencies": {
-    "tslib": "^1.7.1"
-  },
+  "dependencies": {},
```
[the remark this answers is not in the document]
Comment (DarylThayil) [reply lib/msal-core/package.json]: yes, package requires it to be here

File: lib/msal-core/tests/MSAL.spec.ts [hunk 4]
```diff
@@ -840,44 +849,17 @@ describe('Msal', function (): any {
...
-         msal = new UserAgentApplication(config);
-         msal.handleRedirectCallbacks(function(token, tokenType, state) {
+        const config: Configuration = {
+            ...DEFAULT_UAA_CONFIG,
+            cache: {
+                cacheLocation: "localStorage"
+            }
+        };
+        msal = new UserAgentApplication(config);
+        msal.handleRedirectCallbacks(function(token, tokenType, state) {
```
[the remark this answers is not in the document]
Comment (DarylThayil) [reply lib/msal-core/tests/MSAL.spec.ts]: I did not change any of that code,

**Pull request merged**